src/noit_metric_director.c

```diff
@@ -1159,8 +1259,44 @@ handle_kafka_message(void *closure, mtev_rd_kafka_message_t *msg) {
     return MTEV_HOOK_CONTINUE;
   }
   mtev_rd_kafka_message_ref(msg);
-  if(check_duplicate(msg->payload, msg->payload_len) == mtev_false) {
-    handle_metric_buffer(msg->payload, msg->payload_len, 1, NULL);
+  if (!strcasecmp(msg->protocol, "prometheus")) {
+    void *account_id_void_ptr = NULL;
+    void *check_uuid_void_ptr = NULL;
+    if (!mtev_hash_retrieve((mtev_hash_table *)msg->extra_configs, PROMETHEUS_MESSAGE_ACCOUNT_ID_KEY,
```

Comment:
Are these only ever pulled out here?  The string lookup, and parsing is not cheap and we should assure that it is only ever done once.